src/answers-search-bar.js

```diff
@@ -0,0 +1,519 @@
+import ComponentManager from './ui/components/componentmanager';
+import VerticalPagesConfig from './core/models/verticalpagesconfig';
+import { SANDBOX, PRODUCTION, LOCALE, QUERY_SOURCE } from './core/constants';
+import { isValidContext } from './core/utils/apicontext';
+import { urlWithoutQueryParamsAndHash } from './core/utils/urlutils';
+import Filter from './core/models/filter';
+import SearchComponent from './ui/components/search/searchcomponent';
+import QueryUpdateListener from './core/statelisteners/queryupdatelistener';
+import { SEARCH_BAR_COMPONENTS_REGISTRY } from './ui/components/search-bar-only-registry';
+
+/** @typedef {import('./core/services/errorreporterservice').default} ErrorReporterService */
+/** @typedef {import('./core/services/analyticsreporterservice').default} AnalyticsReporterService */
+
+/**
+ * @typedef Services
+ * @property {ErrorReporterService} errorReporterService
+ */
+
+const DEFAULTS = {
+  locale: LOCALE,
+  querySource: QUERY_SOURCE
+};
+
+/**
+ * The class that powers the stand-alone SearchBar integration. Note that this class
+ * follows the singleton pattern.
+ */
+class AnswersSearchBar {
+  constructor () {
+    if (!AnswersSearchBar.setInstance(this)) {
+      return AnswersSearchBar.getInstance();
+    }
+
+    /**
+     * A reference of the renderer to use for the components
+     * This is provided during initialization.
+     * @type {Renderer}
+     */
+    this.renderer = new Renderers.Handlebars();
+
+    /**
+     * A local reference to the component manager
+     * @type {ComponentManager}
+     */
+    this.components = new ComponentManager(SEARCH_BAR_COMPONENTS_REGISTRY);
+
+    /**
+     * A local reference to the core api
+     * @type {Core}
+     */
+    this.core = null;
+
+    /**
+     * A callback function to invoke once the library is ready.
+     * Typically fired after templates are fetched from server for rendering.
+     */
+    this._onReady = function () {};
+
+    /**
+     * @type {boolean}
+     * @private
+     */
+    this._eligibleForAnalytics = false;
+
+    /**
+     * @type {Services}
+     * @private
+     */
+    this._services = null;
+
+    /**
+     * @type {AnalyticsReporterService}
+     * @private
+     */
+    this._analyticsReporterService = null;
+  }
+
+  static setInstance (instance) {
+    if (!this.instance) {
+      this.instance = instance;
+      return true;
+    }
+    return false;
+  }
+
+  static getInstance () {
+    return this.instance;
+  }
+
+  /**
+   * Initializes the stand-alone SearchBar using the provided configuration.
+   *
+   * @param {Object} config The Answers configuration.
+   */
+  init (config) {
+    const parsedConfig = this.parseConfig(config);
+    this.validateConfig(parsedConfig);
+
+    parsedConfig.search = new SearchConfig(parsedConfig.search);
+    parsedConfig.verticalPages = new VerticalPagesConfig(parsedConfig.verticalPages);
+
+    const storage = new Storage({
+      updateListener: (data, url) => {
```

Comment:
would it make sense to git rid of the storage listeners? I don't think we need to do anything when the client's page does back/forward navigation